torchinductor/compile_fx.py

```diff
@@ -190,6 +195,9 @@ def static_input(x):
         return torch.as_strided(buffer, x.size(), x.stride())
 
     assert isinstance(inputs, (list, tuple))
+    # dynamo wraps unspec variable as 0 dim tensor on CPU, need to move to GPU explicitly
+    inputs = [x.to("cuda") if is_unspec_input(x) else x for x in inputs]
```

Comment:
This is inside of ```cudagraphify```, so it will be run only when ```set(graph.device_types) == {"cuda"}```(except 0d tensor). The other place is the same, applied only graph on GPU. I verified on CPU tensors, it works well.